src/shared/loop-util.c

```diff
@@ -685,9 +685,9 @@ int loop_device_make_by_path_at(
                 if (open_flags >= 0 || !(ERRNO_IS_PRIVILEGE(r) || r == -EROFS))
```

Comment:
Not your fault, but the `r = -errno` should be changed to `r = fd` too

Reply:
Good catch!